src/components/NewConversationDialog/NewConversationSetupPage.vue

```diff
@@ -0,0 +1,210 @@
+			{{ t('spreed', 'Conversation visibility') }}
+		</label>
+		<NcCheckboxRadioSwitch :checked.sync="isPublic"
+			type="switch">
+			{{ t('spreed', 'Allow guests to join via link') }}
+		</NcCheckboxRadioSwitch>
+		<div class="new-group-conversation__wrapper">
+			<NcCheckboxRadioSwitch :checked.sync="hasPassword"
+				type="switch"
+				:disabled="!isPublic">
+				<span class="checkbox__label">{{ t('spreed', 'Password protect') }}</span>
+			</NcCheckboxRadioSwitch>
+			<NcPasswordField v-if="hasPassword"
+				v-model="passwordValue"
+				autocomplete="new-password"
+				check-password-strength
+				:placeholder="t('spreed', 'Enter password')"
+				:aria-label="t('spreed', 'Enter password')" />
+		</div>
+		<ListableSettings v-model="listableValue" />
+	</div>
+</template>
+
+<script>
+import { getCapabilities } from '@nextcloud/capabilities'
+
+import NcCheckboxRadioSwitch from '@nextcloud/vue/dist/Components/NcCheckboxRadioSwitch.js'
+import NcPasswordField from '@nextcloud/vue/dist/Components/NcPasswordField.js'
+import NcTextArea from '@nextcloud/vue/dist/Components/NcTextArea.js'
+import NcTextField from '@nextcloud/vue/dist/Components/NcTextField.js'
+
+import ConversationAvatarEditor from '../ConversationSettings/ConversationAvatarEditor.vue'
+import ListableSettings from '../ConversationSettings/ListableSettings.vue'
+
+import { CONVERSATION } from '../../constants.js'
+
+const supportsAvatar = getCapabilities()?.spreed?.features?.includes('avatar')
+
+export default {
+
+	name: 'NewConversationSetupPage',
+
+	components: {
+		ConversationAvatarEditor,
+		ListableSettings,
+		NcCheckboxRadioSwitch,
+		NcPasswordField,
+		NcTextArea,
+		NcTextField,
+	},
+
+	props: {
+		newConversation: {
+			type: Object,
+			required: true,
+		},
+		password: {
+			type: String,
+			required: true,
+		},
+		listable: {
+			type: Number,
+			required: true,
+		}
+	},
+
+	emits: ['update:newConversation', 'update:password', 'update:listable', 'avatar-edited', 'handle-enter'],
+
+	setup() {
+		return { supportsAvatar }
+	},
+
+	computed: {
+		conversationName: {
+			get() {
+				return this.newConversation.displayName
+			},
+			set(event) {
+				this.updateNewConversation({ displayName: event.target.value })
+			},
+		},
+
+		conversationDescription: {
+			get() {
+				return this.newConversation.description
+			},
+			set(event) {
+				this.updateNewConversation({ description: event.target.value })
+			},
+		},
+
+		isPublic: {
+			get() {
+				return this.newConversation.type === CONVERSATION.TYPE.PUBLIC
+			},
+			set(value) {
+				if (value) {
+					this.updateNewConversation({ type: CONVERSATION.TYPE.PUBLIC })
+				} else {
+					this.updateNewConversation({ type: CONVERSATION.TYPE.GROUP, hasPassword: false })
+				}
+			},
+		},
+
+		hasPassword: {
+			get() {
+				return this.newConversation.hasPassword
+			},
+			set(value) {
+				this.updateNewConversation({ hasPassword: value })
+				if (!value) {
+					this.passwordValue = ''
+				}
```

Comment:
This is causing an error `TypeError: Cannot read properties of undefined (reading 'value')`

Reply:
It triggered a computed set, which expects to get event object, not the String
Now it directly updates a value in parent